service/ui_plugin_build_baron.go

```diff
@@ -298,6 +303,70 @@ func (uis *UIServer) bbJiraSearch(rw http.ResponseWriter, r *http.Request) {
 	gimlet.WriteJSON(rw, searchReturnInfo{Issues: tickets, Search: jql, Source: source, FeaturesURL: featuresURL})
 }
 
+// bbFileTicket creates a JIRA ticket for a task with the given test failures.
+func (uis *UIServer) bbFileTicket(w http.ResponseWriter, r *http.Request) {
+	var input struct {
+		TaskId  string   `json:"task"`
+		TestIds []string `json:"tests"`
+	}
+
+	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
+		gimlet.WriteJSONInternalError(w, err.Error())
+	}
+
+	// Find information about the task
+	t, err := task.FindOne(task.ById(input.TaskId))
+	if err != nil {
+		gimlet.WriteJSONInternalError(w, err.Error())
+		return
+	}
+	if t == nil {
+		gimlet.WriteJSONResponse(w, http.StatusNotFound, fmt.Sprintf("task not found for id %s", input.TaskId))
+		return
+	}
+
+	n, err := uis.makeNotification(uis.buildBaronProjects[t.Project].TicketCreateProject, t)
+	if err != nil {
+		gimlet.WriteJSONInternalError(w, err.Error())
+		return
+	}
+	err = uis.queue.Put(units.NewEventNotificationJob(n.ID))
```

Comment:
If we currently return the ticket number created by the user, are we no longer doing this because we are putting the job on the queue? I think `event.LogJiraIssueCreated` takes the ticket number, not the project name as its last argument.

Reply:
Correct, EVG-5261 is open for this. This was broken when ticket creation was transitioned to notifications and this will require some grip changes to support as well

Reply:
Thanks. I scheduled that ticket.